src/main/java/unrc/dose/Proposition.java

```diff
@@ -1,18 +1,417 @@
+        newProposition.set("user_id", currentUser.getInteger("id"));
+        newProposition.set("challenge_id", currentChallenge.getInteger("id"));
+        newProposition.set("source", currentChallenge.get("source"));
+        newProposition.set("isSolution", false);
+        newProposition.set("distance", 0);
+        newProposition.set("cantTestPassed", 0);
+        newProposition.saveIt();
+        return newProposition;
+    }
+
+    /**
+     * Get the solutions for a challenge by a specific user..
+     * @param userId reference the id of a user
+     * @param challengeId reference the id of a challenge
+     * @return List of proposition for user in challenge
+     */
+    public static LazyList<Proposition> getChallengeSolutionsByUser(
+            final Integer userId, final Integer challengeId) {
+        return Proposition.where("user_id = ? and challenge_id = ? "
+                + "and isSolution = ?", userId, challengeId, true);
+    }
+
+    /**
+     * Save a user's solution in a challenge.
+     * @param proposedCode reference the new code of the proposed solution
+     * @param distanceObtained reference the new distance obtained
+     */
+    public void saveSolution(
+        final String proposedCode,
+        final int distanceObtained) {
+        this.setSource(proposedCode);
+        this.set("distance", distanceObtained);
+        this.set("isSolution", true);
+    }
+
+    /**
+     * save proposed solution in a challenge.
+     * @param sourceCurrent reference the new code of the proposed solution
+     */
+    public void saveProposedSolution(final String sourceCurrent) {
+        this.set("source", sourceCurrent);
+    }
+
+    /**
+     * This method find the proposition that is not a solution yet.
+     * @param userId represents the user
+     * @param challengeId represent the challenge
+     * @return null if there is no proposition or the proposition,
+     *  or the proposition found.
+     */
+    public static LazyList<Proposition> getUnsubmittedChallengePropByUser(
+        final int userId,
+        final int challengeId) {
+        return Proposition.where("user_id = ? and challenge_id = ? and "
+                + "isSolution = ?", userId, challengeId, false);
+    }
+
+    /**
+     * This method find the proposition that is solution
+     * for one specific challenge.
+     * @param challengerId represent the challenge.
+     * @return list the solution for one challenge.
+     */
+    public static LazyList<Proposition> getAllSolutionsForChallenge(
+        final int challengerId) {
+        return Proposition.where("challenge_id = ? and isSolution = ?",
+                challengerId, true);
+    }
+
+    /**
+     * This method calculates the editing distance of a Proposition.
+     * @param p represents a Proposition
+     * @return the editing distance of both string
+     */
+    public static int getDistanceProposition(final Proposition p) {
```

Comment:
The proposition's distance to the original code is an attribute of `Proposition` so there should not be a static method that calculate this for a proposition.